libfreeipmi/driver/ipmi-inteldcmi-driver.c

```diff
@@ -444,6 +445,19 @@ ipmi_inteldcmi_ctx_io_init (ipmi_inteldcmi_ctx_t ctx)
       goto cleanup;
     }
 
+  flags = fcntl(ctx->device_fd, F_GETFD);
+  if (flags == -1)
+    {
+      OPENIPMI_ERRNO_TO_OPENIPMI_ERRNUM (ctx, errno);
+      goto cleanup;
```

Comment:
you also seem to have cut & paste the "OPENIPMI_ERRNO_TO..." to all of the code below.  It should be the driver name appropriately.